hooks/use-search.tsx

```
@@ -0,0 +1,302 @@
+    // Use setTimeout to move search off the main thread
+    setTimeout(() => {
+      try {
+        if (!fuse.current) {
+          // If no query but we have filters, just filter the data
+          if (!params.query && (params.chapter || params.narrator)) {
+            const filtered = data.filter(
+              (item) =>
+                (!params.chapter || item.chapter === params.chapter) &&
+                (!params.narrator ||
+                  (item.narrator_name &&
+                    item.narrator_name.includes(params.narrator))),
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)